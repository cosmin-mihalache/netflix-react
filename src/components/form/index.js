import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
