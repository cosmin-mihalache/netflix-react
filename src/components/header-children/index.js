import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
