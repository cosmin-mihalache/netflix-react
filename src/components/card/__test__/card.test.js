import React from 'react';
import {
  render,
  fireEvent,
  getByText,
  getByAltText,
  queryAllByTestId,
} from '@testing-library/react';
import { Card, Player } from '../../../components';

const category = 'series';
const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'amanda-knox',
        description:
          'Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.',
        id: '1ab5cec5-bfe7-4f6c-81b2-bb4638aeb7b3',
        title: 'Amanda Knox',
        docId: '4miTnQ8VItOKKcD0Eyou',
      },
    ],
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description:
          'Social misfit Juno protects herself with a caustic wit, but her unplanned pregnancy has the teen getting more involved in the lives of her babys adoptive parents than she expected.',
        maturity: '6',
        id: '3ef8d1dc-2a04-4c3d-a3da-67d5a268969e',
        slug: 'juno',
        docId: 'FJuauc5Ja3jzi2nH50ar',
      },
    ],
  },
];

describe('<Card />', () => {
  it('renders the <Card /> with populated data', () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(getByText('Documentaries')).toBeTruthy();
    expect(getByText('Amanda Knox')).toBeTruthy();
    expect(
      getByText(
        'Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.'
      )
    ).toBeTruthy();

    expect(getByText('Feel Good')).toBeTruthy();
    expect(getByText('Juno')).toBeTruthy();
    expect(
      getByText(
        'Social misfit Juno protects herself with a caustic wit, but her unplanned pregnancy has the teen getting more involved in the lives of her babys adoptive parents than she expected.'
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Card /> and toggles the card feature', () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image
                    data-testid={`${item.slug}-item-feature`}
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('18')).toBeFalsy();
    fireEvent.click(getByTestId('amanda-knox-item-feature'));
    expect(queryByText('18')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('18')).toBeFalsy();

    expect(queryByText('PG')).toBeFalsy();
    fireEvent.click(getByTestId('juno-item-feature'));
    expect(queryByText('PG')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('PG')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
