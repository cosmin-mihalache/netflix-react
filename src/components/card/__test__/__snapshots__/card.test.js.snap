// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the <Card /> and toggles the card feature 1`] = `
<div
  class="card__Group-sc-1li1lye-2 ONTqS"
>
  <div
    class="card__Container-sc-1li1lye-1 gvEZAM"
  >
    <p
      class="card__Title-sc-1li1lye-0 hVUIak"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-1li1lye-12 cwuFd"
    >
      <div
        class="card__Item-sc-1li1lye-13 iknsgI"
      >
        <img
          class="card__Image-sc-1li1lye-14 kCnvoJ"
          data-testid="amanda-knox-item-feature"
          src="/images/series/documentaries/amanda-knox/small.jpg"
        />
        <div
          class="card__Meta-sc-1li1lye-11 eLjeNd"
        >
          <p
            class="card__SubTitle-sc-1li1lye-3 dGrqJQ"
          >
            Amanda Knox
          </p>
          <p
            class="card__Text-sc-1li1lye-4 iHlEut"
          >
            Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-1li1lye-1 gvEZAM"
  >
    <p
      class="card__Title-sc-1li1lye-0 hVUIak"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-1li1lye-12 cwuFd"
    >
      <div
        class="card__Item-sc-1li1lye-13 iknsgI"
      >
        <img
          class="card__Image-sc-1li1lye-14 kCnvoJ"
          data-testid="juno-item-feature"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-1li1lye-11 eLjeNd"
        >
          <p
            class="card__SubTitle-sc-1li1lye-3 dGrqJQ"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-1li1lye-4 iHlEut"
          >
            Social misfit Juno protects herself with a caustic wit, but her unplanned pregnancy has the teen getting more involved in the lives of her babys adoptive parents than she expected.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> renders the <Card /> with populated data 1`] = `
<div
  class="card__Group-sc-1li1lye-2 ONTqS"
>
  <div
    class="card__Container-sc-1li1lye-1 gvEZAM"
  >
    <p
      class="card__Title-sc-1li1lye-0 hVUIak"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-sc-1li1lye-12 cwuFd"
    >
      <div
        class="card__Item-sc-1li1lye-13 iknsgI"
      >
        <img
          class="card__Image-sc-1li1lye-14 kCnvoJ"
          src="/images/series/documentaries/amanda-knox/small.jpg"
        />
        <div
          class="card__Meta-sc-1li1lye-11 eLjeNd"
        >
          <p
            class="card__SubTitle-sc-1li1lye-3 dGrqJQ"
          >
            Amanda Knox
          </p>
          <p
            class="card__Text-sc-1li1lye-4 iHlEut"
          >
            Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-1li1lye-1 gvEZAM"
  >
    <p
      class="card__Title-sc-1li1lye-0 hVUIak"
    >
      Feel Good
    </p>
    <div
      class="card__Entities-sc-1li1lye-12 cwuFd"
    >
      <div
        class="card__Item-sc-1li1lye-13 iknsgI"
      >
        <img
          class="card__Image-sc-1li1lye-14 kCnvoJ"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="card__Meta-sc-1li1lye-11 eLjeNd"
        >
          <p
            class="card__SubTitle-sc-1li1lye-3 dGrqJQ"
          >
            Juno
          </p>
          <p
            class="card__Text-sc-1li1lye-4 iHlEut"
          >
            Social misfit Juno protects herself with a caustic wit, but her unplanned pregnancy has the teen getting more involved in the lives of her babys adoptive parents than she expected.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
